mentat/src/models/operation.rs

```diff
@@ -2,25 +2,59 @@ use indexmap::IndexMap;
-    /// Amount is some Value of a Currency. It is considered invalid to specify a Value without a Currency.
-    pub amount: Option<Amount>,
-    /// CoinChange is used to represent a change in state of a some coin identified by a coin_identifier. This object is part of the Operation model and must be populated for UTXO-based blockchains. Coincidentally, this abstraction of UTXOs allows for supporting both account-based transfers and UTXO-based transfers on the same blockchain (when a transfer is account-based, don't populate this model).
-    pub coin_change: Option<CoinChange>,
-    /// Any additional information related to the currency itself. For example, it would be useful to populate this object with the contract address of an ERC-20 token.
+    pub type_:                String,
+    /// Status is the network-specific status of the operation. Status is not
+    /// defined on the transaction object because blockchains with smart
+    /// contracts may have transactions that partially apply (some operations
+    /// are successful and some are not). Blockchains with atomic transactions
+    /// (all operations succeed or all operations fail) will have the same
+    /// status for each operation. On-chain operations (operations retrieved in
+    /// the /block and /block/transaction endpoints) MUST have a populated
+    /// status field (anything on-chain must have succeeded or failed). However,
+    /// operations provided during transaction construction (often times called
+    /// "intent" in the documentation) MUST NOT have a populated status field
+    /// (operations yet to be included on-chain have not yet succeeded or
+    /// failed).
+    pub status:               Option<String>,
+    /// The account_identifier uniquely identifies an account within a network.
+    /// All fields in the account_identifier are utilized to determine this
+    /// uniqueness (including the metadata field, if populated).
+    pub account:              Option<AccountIdentifier>,
+    /// Amount is some Value of a Currency. It is considered invalid to specify
+    /// a Value without a Currency.
+    pub amount:               Option<Amount>,
+    /// CoinChange is used to represent a change in state of a some coin
+    /// identified by a coin_identifier. This object is part of the Operation
+    /// model and must be populated for UTXO-based blockchains. Coincidentally,
+    /// this abstraction of UTXOs allows for supporting both account-based
+    /// transfers and UTXO-based transfers on the same blockchain (when a
+    /// transfer is account-based, don't populate this model).
+    pub coin_change:          Option<CoinChange>,
+    /// Any additional information related to the currency itself. For example,
+    /// it would be useful to populate this object with the contract address of
+    /// an ERC-20 token.
     #[serde(default)]
-    pub metadata: IndexMap<String, Value>,
+    pub metadata:             IndexMap<String, Value>,
 }
```

Comment:
i have to say im not the biggest fan of the auto-aligning struct fields

Reply:
We can disable it if you like. I usually like it, and I'm used to it since it's the standard of most other langs' style guides.

Reply:
I do like it, but only if struct fields are adjacent. When they're broken up by comments IMO it doesn't really look that nice